src/components/SingleName/Name.js

```diff
@@ -32,36 +35,60 @@ function isRegistrationOpen(available, parent, isDeedOwner) {
 }
 
 function isDNSRegistrationOpen(domain) {
-  const nameArray = domain.name.split('.')
-  if (nameArray.length !== 2 || nameArray[1] === 'eth') {
+  const nameArray = domain.name?.split('.')
+  if (nameArray?.length !== 2 || nameArray?.[1] === 'eth') {
     return false
   }
   return domain.isDNSRegistrar && domain.owner === EMPTY_ADDRESS
 }
 
 function isOwnerOfDomain(domain, account) {
   if (domain.owner !== EMPTY_ADDRESS && !domain.available) {
-    return domain.owner.toLowerCase() === account.toLowerCase()
+    return domain.owner?.toLowerCase() === account?.toLowerCase()
   }
   return false
 }
 
-function isOwnerOfParentDomain(domain, account) {
-  if (domain.parentOwner !== EMPTY_ADDRESS) {
-    return domain.parentOwner.toLowerCase() === account.toLowerCase()
+const NAME_REGISTER_DATA_WRAPPER = gql`
+  query nameRegisterDataWrapper @client {
+    accounts
+    networkId
   }
-  return false
+`
+
+export const useRefreshComponent = () => {
```

Comment:
Is this what force refreshes when network has changed? I'm sure I'm the reason this is here (although you look like you've abstracted to a hook), but maybe there's a better way?